pkg/store/limiter.go

```diff
@@ -82,16 +86,89 @@ func NewChunksLimiterFactory(limit uint64) ChunksLimiterFactory {
 	}
 }
 
-// NewSeriesLimiterFactory makes a new NewSeriesLimiterFactory with a static limit.
+// NewSeriesLimiterFactory makes a new SeriesLimiterFactory with a static limit.
 func NewSeriesLimiterFactory(limit uint64) SeriesLimiterFactory {
 	return func(failedCounter prometheus.Counter) SeriesLimiter {
 		return NewLimiter(limit, failedCounter)
 	}
 }
 
-// NewSeriesLimiterFactory makes a new NewSeriesLimiterFactory with a static limit.
+// NewBytesLimiterFactory makes a new BytesLimiterFactory with a static limit.
 func NewBytesLimiterFactory(limit units.Base2Bytes) BytesLimiterFactory {
 	return func(failedCounter prometheus.Counter) BytesLimiter {
 		return NewLimiter(uint64(limit), failedCounter)
 	}
 }
+
+type RateLimits struct {
+	SeriesPerRequest  uint64
+	SamplesPerRequest uint64
+}
```

Comment:
Would be nice to explain in a comment here what's the purpose of this type and/or how the limit works.

For example, I wouldn't even call this type of limiting "rate limiting" because it's not happening over a period of time, like 30 series per second. this limiting is, as the name of the struct fields show, per request limits.

Reply:
I added a comment on this type to explain how limits are applied. I also renamed the server to `limitedServer` and dropped the `rate` from it.